Configure formatting rules with ESLint and Prettier

pedro-m-dietrich opened the pull request:
The TypeScript code will follow the rules defined by the **ESLint** and **Prettier**.

pedro-m-dietrich committed:
Setup ESLint for TypeScript

pedro-m-dietrich committed:
Use Prettier for code formatting

Prettier has been configured and the TypeScript code has been updated to
obey the new formatting rules.

alanraso commented on eslint.config.mjs:
```diff
@@ -0,0 +1,31 @@
+import tseslint from 'typescript-eslint';
+import eslintPluginPrettierRecommended from 'eslint-plugin-prettier/recommended';
+
+export default tseslint.config(
+    eslint.configs.recommended,
+    ...tseslint.configs.strict,
+    ...tseslint.configs.stylistic,
```
Lá no index a identação está usando 2 espaços, mas aqui e no `.prettierrc.json` está usando 4. Sugiro configurar seu VSCode pra 2 espaços por padrão, assim esses arquivos que não entram nas regras do lint/prettier ficam dentro desse padrão tbm.

pedro-m-dietrich committed:
Fix tab size on all files

Some files are not checked by ESLint or Prettier. Despite that, to keep
the same pattern in all files in the project, these unchecked files will
also use a tab size of 2.
